### DISCUSSION TITLE
*Help me! Vosk and Ffmpeg*

### DISCUSSION THREAD

**Discussion by sandroamiridze** [community/community#133978] [Programming Help] [answered]
### Body

when i run this, it quits and nothing happens! its voice to text program.

#!/bin/sh

ffmpeg -f pulse -i alsa_input.pci-0000_00_1b.0.analog-stereo -ac 1 -ar 44100 -t 5 ~/.cache/audios.wav >/dev/null 2>&1 vosk-transcriber -m ~/.local/share/kdenlive/speechmodels/vosk-model-small-en-us-0.15 -i ~/.cache/audios.wav -o ~/.cache/transcript.txt >/dev/null 2>&1
cat ~/.cache/transcript.txt

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by Faiaz3k** [comment, marked as answer]
use this script for debugging
#!/bin/sh

# Record audio
echo "Recording audio..."
ffmpeg -f pulse -i alsa_input.pci-0000_00_1b.0.analog-stereo -ac 1 -ar 44100 -t 5 ~/.cache/audios.wav
if [ $? -ne 0 ]; then
  echo "FFmpeg recording failed"
  exit 1
fi
echo "Audio recorded successfully."

# Check if audio file exists
if [ ! -f ~/.cache/audios.wav ]; then
  echo "Audio file not found"
  exit 1
fi

# Transcribe audio
echo "Transcribing audio..."
vosk-transcriber -m ~/.local/share/kdenlive/speechmodels/vosk-model-small-en-us-0.15 -i ~/.cache/audios.wav -o ~/.cache/transcript.txt
if [ $? -ne 0 ]; then
  echo "Vosk transcription failed"
  exit 1
fi
echo "Transcription completed successfully."

# Check if transcript file exists
if [ ! -f ~/.cache/transcript.txt ]; then
  echo "Transcript file not found"
  exit 1
fi

# Display transcript
cat ~/.cache/transcript.txt

**Reply by sandroamiridze** [reply]
thank you so much my brother <3
